sdk/identity/azure-identity/azure/identity/_credentials/azure_cli.py

```diff
@@ -136,10 +140,14 @@ def get_token_info(self, *scopes: str, options: Optional[TokenRequestOptions] =
         :rtype: ~azure.core.credentials.AccessTokenInfo
         :return: An AccessTokenInfo instance containing information about the token.
 
-        :raises ~azure.identity.CredentialUnavailableError: the credential was unable to invoke the Azure CLI.
+        :raises ~azure.identity.CredentialUnavailableError: the credential was either unable to invoke the Azure CLI
+          or a claims challenge was provided.
         :raises ~azure.core.exceptions.ClientAuthenticationError: the credential invoked the Azure CLI but didn't
           receive an access token.
         """
+        claims_value = options.get("claims") if options else None
+        if claims_value and claims_value.strip():
+            raise CredentialUnavailableError(f"Failed to get token. Run az login --claims-challenge {claims_value}")
```

Comment:
What should be the format of encoding here? Right now, the credential receives the `claims` as a JSON string, and we end up passing this unquoted, which might not work for passing the argument through the shell environment.